lib/spore/environment.rb

```diff
@@ -17,11 +17,20 @@ def read
     end
 
     def apply
-      each { |k, v| ENV[k] ||= v }
+      each { |k, v| ENV[k] ||= maybe_indirect(v) }
     end
 
     def apply!
-      each { |k, v| ENV[k] = v }
+      each { |k, v| ENV[k] = maybe_indirect(v) }
+    end
+
+    private
+
+    # Indirect values point to other environment variables and start with two
+    # tildes (~~).
```

Comment:
No bother, I'll take the same approach and while at it fix the bug in dotenv's code where unbalanced curly braces around the environment variable would still trigger a substitution.